example/hessian.cpp

```diff
@@ -0,0 +1,259 @@
+#include <boost/gil/image.hpp>
+#include <boost/gil/image_view.hpp>
+#include <boost/gil/image_processing/numeric.hpp>
+#include <boost/gil/image_processing/hessian.hpp>
+#include <boost/gil/extension/io/png.hpp>
+#include <vector>
+#include <functional>
+#include <set>
+#include <iostream>
+#include <fstream>
+
+namespace gil = boost::gil;
+
+// some images might produce artifacts
+// when converted to grayscale,
+// which was previously observed on
+// canny edge detector for test input
+// used for this example
+gil::gray8_image_t to_grayscale(gil::rgb8_view_t original)
+{
+    gil::gray8_image_t output_image(original.dimensions());
+    auto output = gil::view(output_image);
+    constexpr double max_channel_intensity = (std::numeric_limits<std::uint8_t>::max)();
+    for (long int y = 0; y < original.height(); ++y)
+    {
+        for (long int x = 0; x < original.width(); ++x)
+        {
+            // scale the values into range [0, 1] and calculate linear intensity
+            auto& p = original(x, y);
+            double red_intensity = p.at(std::integral_constant<int, 0>{})
+                / max_channel_intensity;
+            double green_intensity = p.at(std::integral_constant<int, 1>{})
+                / max_channel_intensity;
+            double blue_intensity = p.at(std::integral_constant<int, 2>{})
+                / max_channel_intensity;
+            auto linear_luminosity = 0.2126 * red_intensity
```

Comment:
It would be useful to provide a literature reference to explain the choice of those three weighing factors in the luminosity computation.

Reply:
I have added a link to formula used at the top of the function

Reply:
Sure, thanks.